src/server/db/mod.rs

```diff
@@ -0,0 +1,153 @@
+mod migrations;
+
+use dirs::WORK_DIR;
+use errors::*;
+use r2d2::{CustomizeConnection, Pool};
+use r2d2_sqlite::SqliteConnectionManager;
+use rusqlite::{Connection, Row, Transaction};
+use rusqlite::types::ToSql;
+use std::sync::Arc;
+use tempfile::NamedTempFile;
+
+static DATABASE_PATH: &'static str = "server.db";
+
+#[derive(Debug)]
+struct ConnectionCustomizer;
+
+impl CustomizeConnection<Connection, ::rusqlite::Error> for ConnectionCustomizer {
+    fn on_acquire(&self, conn: &mut Connection) -> ::std::result::Result<(), ::rusqlite::Error> {
+        conn.execute("PRAGMA foreign_keys = ON;", &[])?;
+        Ok(())
+    }
+}
+
+#[derive(Clone)]
+pub struct Database {
+    pool: Pool<SqliteConnectionManager>,
+    // The tempfile is stored here to drop it after all the connections are closed
+    tempfile: Option<Arc<NamedTempFile>>,
+}
+
+impl Database {
+    pub fn open() -> Result<Self> {
+        let path = WORK_DIR.join(DATABASE_PATH);
+        Database::new(SqliteConnectionManager::file(path), None)
+    }
+
+    #[cfg(test)]
+    pub fn temp() -> Result<Self> {
+        let tempfile = NamedTempFile::new()?;
+        Database::new(
+            SqliteConnectionManager::file(tempfile.path()),
```

Comment:
Why not just `SqliteConnectionManager::memory`?

Reply:
https://github.com/ivanceras/r2d2-sqlite/issues/1

Reply:
To make things clear, in-memory SQLite databases don't really work with a connection pool since each connection is a different database.